Exiled.API/Features/Player.cs

```diff
@@ -1500,16 +1500,37 @@ public void Heal(float amount, bool overrideMaxHealth = false)
         /// <summary>
         /// Kills the player.
         /// </summary>
-        /// <param name="deathReason">The reason the player has been killed.</param>
-        /// <param name="cassieAnnouncement">The cassie announcement to make upon death.</param>
-        public void Kill(string deathReason, string cassieAnnouncement = "")
+        /// <param name="damageHandlerBase">The <see cref="DamageHandlerBase"/> used to kill.</param>
+        public void Kill(DamageHandlerBase damageHandlerBase)
         {
-            if (Side != Side.Scp && !string.IsNullOrEmpty(cassieAnnouncement))
-                Cassie.Message(cassieAnnouncement);
+            if (Side != Side.Scp && !string.IsNullOrEmpty(damageHandlerBase.CassieDeathAnnouncement.Announcement))
+                Cassie.Message(damageHandlerBase.CassieDeathAnnouncement.Announcement);
 
-            ReferenceHub.playerStats.KillPlayer(new CustomReasonDamageHandler(deathReason, float.MaxValue, cassieAnnouncement));
+            ReferenceHub.playerStats.KillPlayer(damageHandlerBase);
         }
 
+        /// <summary>
+        /// Kills the player.
+        /// </summary>
+        /// <param name="handler">The <see cref="DamageHandler"/> used to kill.</param>
+        public void Kill(DamageHandler handler) => Kill(handler.Base);
+
+        /// <summary>
+        /// Kills the player.
+        /// </summary>
+        /// <param name="damageType">The <see cref="DamageType">damage type</see> the player has been killed with.</param>
+        /// <param name="killer">The <see cref="Player"/> who killed player.</param>
+        /// <param name="cassieAnnouncement">The cassie announcement to make upon death.</param>
+        public void Kill(DamageType damageType = DamageType.Unknown, Player killer = null, string cassieAnnouncement = "") => Kill(new ExiledDamageHandler(killer, float.MaxValue, cassieAnnouncement, DamageHandler.TranslationConversion.FirstOrDefault(k => k.Value == damageType).Key.LogLabel));
+
+        /// <summary>
+        /// Kills the player.
+        /// </summary>
+        /// <param name="deathReason">The reason the player has been killed.</param>
+        /// <param name="killer">The <see cref="Player"/> who killed player.</param>
+        /// <param name="cassieAnnouncement">The cassie announcement to make upon death.</param>
+        public void Kill(string deathReason, Player killer = null, string cassieAnnouncement = "") => Kill(new ExiledDamageHandler(killer, float.MaxValue, cassieAnnouncement, deathReason));
```

Comment:
Same as above + the `deathReason` does not work.

There are 2 solutions:
1) Fixing ExiledDamageHandler
2) Using `CustomReasonDamageHandler`. The problem with that is second constructor, which will always cause NRE (NW issue, they already know(